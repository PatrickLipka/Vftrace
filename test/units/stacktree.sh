#!/bin/bash
set -x
test_name=stacktree
output_file=${test_name}.out
ref_file=${srcdir}/ref_output/${test_name}.out

rm -f ${output_file}

if [ "x$HAS_MPI" == "xYES" ]; then
   ${MPI_EXEC} ${MPI_OPTS} ${NP} 1 ./${test_name} > ${output_file}
else
   ./${test_name} > ${output_file}
fi

last_success=$?

# We need to filter out the Address lines because they are
# not reproducible. 
# diff file1 <(expression) does not work when called with make check
if [ $last_success == 0 ]; then
  diff ${output_file} ${ref_file}
else
  exit  ${last_success}
fi

