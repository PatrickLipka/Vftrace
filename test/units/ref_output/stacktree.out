init (0): id=0, calls: 0, incl_time/s: 0.000000
  func0 (1e240): id=1, calls: 0, incl_time/s: 0.000000
    func3 (1e243): id=4, calls: 0, incl_time/s: 0.000000
    func4 (1e244): id=5, calls: 0, incl_time/s: 0.000000
      func5 (1e245): id=6, calls: 0, incl_time/s: 0.000000
  func1 (1e241): id=2, calls: 0, incl_time/s: 0.000000
    func2 (1e242): id=3, calls: 0, incl_time/s: 0.000000
